File: api/proxy.py
import json
from flask import Flask, request
import google.generativeai as genai

app = Flask(__name__)

# 配置 Gemini 2.5 API Key
genai.configure(api_key="AIzaSyA7UCf2pgHX8dMlqDqxbbkRNweUSmfy9B8")

@app.route("/proxy", methods=["POST"])
def proxy():
    try:
        data = request.json
        prompt = data.get("prompt", "")
        if not prompt:
            return json.dumps({"error": "No prompt provided"}), 400

        response = genai.generate_text(
            model="gemini-2.5",
            prompt=prompt
        )
        return json.dumps({"text": response.text})
    except Exception as e:
        return json.dumps({"error": str(e)}), 500

File: api/proxy.js
export default async function handler(req, res) {
  try {
    // 从请求体取出用户传过来的 prompt
    const { prompt } = req.body;

    // 调用 Google Generative AI API
    const response = await fetch("https://generativelanguage.googleapis.com/v1beta/models/gemini-pro:generateContent?key=" + process.env.GOOGLE_API_KEY, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        contents: [{ parts: [{ text: prompt }]}]
      }),
    });

    const data = await response.json();
    res.status(200).json(data);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
}
